src/app/modules/submission/multirespondent-submission/multirespondent-submission.middleware.ts

```diff
@@ -127,41 +164,68 @@ export const createFormsgAndRetrieveForm = async (
         error,
       })
     })
-    .map((featureFlags) => {
+    .andThen((featureFlags) => {
       // Step 2b: Set formsg.featureFlags
       formsg.featureFlags = featureFlags
-
-      // Step 3: Retrieve form
-      return FormService.retrieveFullFormById(formId)
+      // Step 3: Retrieve mrf submission if exists
+      return retrieveMultirespondentSubmissionIfExists(submissionId)
         .mapErr((error) => {
-          logger.warn({
-            message: 'Failed to retrieve form from database',
+          logger.error({
+            message: 'Error occurred whilst retrieving mrf submission',
             meta: logMeta,
             error,
           })
-          const { errorMessage, statusCode } = mapRouteError(error)
+          const { statusCode, errorMessage } = mapRouteError(error)
           return res.status(statusCode).json({ message: errorMessage })
         })
-        .map((formDef) =>
-          // Step 4a: Check form is multirespondent form
-          checkFormIsMultirespondent(formDef)
+        .map((mrfSubmission) => {
+          formsg.mrfSubmission = mrfSubmission
+          return mrfSubmission
+        })
```

Comment:
we're moving the mrf submission retrieval up so that we can include the snapshotted form definition.